pkg/collector/corechecks/system/cpu/cpu/cpu.go

```diff
@@ -83,36 +125,39 @@ func (c *Check) Run() error {
 		sender.Gauge("system.cpu.stolen", stolen*toPercent, "", nil)
 		sender.Gauge("system.cpu.guest", guest*toPercent, "", nil)
 	}
-
-	sender.Gauge("system.cpu.num_cores", c.nbCPU, "", nil)
-	sender.Commit()
-
 	c.lastNbCycle = nbCycle
 	c.lastTimes = t
 	return nil
 }
 
-// Configure the CPU check
-func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
-	err := c.CommonConfigure(senderManager, initConfig, data, source)
+func (c *Check) reportCpuMetricsPerCpu(sender sender.Sender) (err error) {
+	cpuTimes, err := getCpuTimes(true)
 	if err != nil {
+		log.Errorf("could not retrieve cpu stats: %s", err.Error())
 		return err
 	}
-	// NOTE: This runs before the python checks, so we should be good, but cpuInfo()
-	//       on windows initializes COM to the multithreaded model. Therefore,
-	//       if a python check has run on this native windows thread prior and
-	//       CoInitialized() the thread to a different model (ie. single-threaded)
-	//       This will cause cpuInfo() to fail.
-	info, err := cpuInfoFunc()
-	if err != nil {
-		return fmt.Errorf("cpu.Check: could not query CPU info")
-	}
-	for _, i := range info {
-		c.nbCPU += float64(i.Cores)
+	log.Debugf("getCpuTimes(true): %s", cpuTimes)
+	for _, t := range cpuTimes {
+		tags := []string{fmt.Sprintf("core:%s", t.CPU)}
+		sender.Gauge("system.cpu.user.total", t.User, "", tags)
+		sender.Gauge("system.cpu.nice.total", t.Nice, "", tags)
+		sender.Gauge("system.cpu.system.total", t.System, "", tags)
+		sender.Gauge("system.cpu.idle.total", t.Idle, "", tags)
+		sender.Gauge("system.cpu.iowait.total", t.Iowait, "", tags)
+		sender.Gauge("system.cpu.irq.total", t.Irq, "", tags)
+		sender.Gauge("system.cpu.softirq.total", t.Softirq, "", tags)
+		sender.Gauge("system.cpu.steal.total", t.Steal, "", tags)
+		sender.Gauge("system.cpu.guest.total", t.Guest, "", tags)
+		sender.Gauge("system.cpu.guestnice.total", t.GuestNice, "", tags)
 	}
 	return nil
 }
 
+// Configure the CPU check
+func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
```

Comment:
Why do we override the `Configure` method?

Reply:
It was needed with the previous code, not now. Removed